### PR TITLE
TSS-106 - Progress update barrier details

### PR DESCRIPTION
**Pull request by jamespacileo**

### CODE REVIEW

**Comment by Santino-Trade** [comment]
The unit tests need fixing

**Review by Santino-Trade — APPROVED** [review]

File: api/metadata/utils.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def import_api_results(endpoint):
     if response.ok:
         return response.json()
 
+    # TODO: log failed request to DH metadata as it fails silently
```
[the remark this answers is not in the document]
Comment (jamespacileo) [reply api/metadata/utils.py:54]: this was something that we spotted when we were doing the onboarding.

I've removed the line. Unrelated to my ticket.

**Review by Santino-Trade — APPROVED** [review]

File: api/barriers/serializers/base.py [hunk 2]
```diff
@@ -67,6 +70,12 @@ class BarrierSerializerBase(
     wto_profile = WTOProfileField(required=False)
     government_organisations = OrganisationsField(required=False)
 
+    progress_status = ChoiceField(choices=PROGRESS_UPDATE_CHOICES, required=False)
```
[the remark this answers is not in the document]
Comment (jamespacileo) [reply api/barriers/serializers/base.py]: just saw this comment. the lines were removed

File: api/barriers/serializers/base.py [hunk 3]
```diff
@@ -86,6 +95,7 @@ class Meta:
             "strategic_assessments",
             "unarchived_by",
             "unarchived_on",
+            "progress_updates",
```
[the remark this answers is not in the document]
Comment (jamespacileo) [reply api/barriers/serializers/base.py:91]: The progress updates get updated via its own API endpoint.

The progress updates are only for view here

File: api/barriers/models.py [hunk 4]
```diff
@@ -277,6 +302,18 @@ class Barrier(FullyArchivableMixin, BaseModel):
         blank=True,
         null=True,
     )
+    progress_status = models.CharField(
```
[the remark this answers is not in the document]
Comment (jamespacileo) [reply api/barriers/models.py]: agreed. they're gone

File: api/barriers/migrations/0116_auto_20220304_1120.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Generated by Django 3.2.7 on 2022-03-04 11:20
```
[the remark this answers is not in the document]
Comment (jamespacileo) [reply api/barriers/migrations/0116_auto_20220304_1120.py:1]: removed

**Pull request merged**